cg/apps/lims/orderform.py

```diff
@@ -120,14 +130,59 @@ def get_project_type(document_title: str, parsed_samples: List) -> str:
         if len(analyses) != 1:
             raise OrderFormError(f"mixed 'Data Analysis' types: {', '.join(analyses)}")
 
-        project_type = analyses.pop()
-        if "mip" in project_type and "balsamic" in project_type:
-            raise OrderFormError(f"mixed 'Data Analysis' types: {project_type}")
+        analysis = analyses.pop()
+
+        if analysis == "no analysis":
+            project_type = "fastq"
+        else:
+            project_type = analysis
 
     return project_type
 
 
-def expand_case(case_id: str, parsed_case: dict) -> tuple:
+def get_data_delivery(samples: [dict], project_type: OrderType) -> str:
+    """Determine the order_data delivery type."""
+
+    data_deliveries = parse_data_deliveries(samples)
+
+    if len(data_deliveries) > 1:
+        raise OrderFormError(f"mixed 'Data Delivery' types: {', '.join(data_deliveries)}")
+
+    data_delivery = data_deliveries.pop()
+
+    if data_delivery == NO_VALUE:
+        if project_type == OrderType.METAGENOME:
+            return str(DataDelivery.FASTQ)
+        elif project_type == OrderType.FASTQ:
+            return str(DataDelivery.FASTQ)
+        elif project_type == OrderType.RML:
+            return str(DataDelivery.FASTQ)
+        elif project_type == OrderType.MIP_RNA:
+            return str(DataDelivery.ANALYSIS_FILES)
+        elif project_type == OrderType.FLUFFY:
+            return str(DataDelivery.NIPT_VIEWER)
+
+        return ""
+
+    try:
+        return str(DataDelivery(data_delivery))
+    except ValueError:
+        raise OrderFormError(f"Unsupported Data Delivery: {data_delivery}")
+
+
+def parse_data_deliveries(samples: [dict]) -> set:
+
+    data_deliveries = set()
+    for sample in samples:
+        data_delivery = (
+            (sample.get("data_delivery", NO_VALUE) or NO_VALUE).lower().replace(" ", "-")
+        )
+        data_deliveries.add(data_delivery)
+
+    return data_deliveries
+
+
+def expand_case(case_id: str, parsed_case: dict, data_delivery: DataDelivery) -> tuple:
```

Comment:
I don't follow here. Function is named `expand_case` but it looks like ot both alters something and returns something. Could it do one of those?

Reply:
I'm not following exactly.. Yes it returns something, but whats being altered?

Reply:
If it does return something and not actually change something (like the name indicates) I suggest it should be named `get_this_or_that` or something similar. Do you follow?
`expand_case` indicates that you send in an abject and it get's expanded.